cerebrium/cortex/advanced-functionality/faster-model-loading.mdx

```diff
@@ -0,0 +1,282 @@
+---
+title: Decrease Model Loading Time
+description: Decrease the time it takes to load your model from storage into GPU
+---
+
+One of the biggest contributions to the startup time of your model is the time it takes to load your model from storage into GPU memory. For example, in larger models of 20B+ parameters, it can take >40s for your model to be loaded using a normal huggingface load, even with the 2GB/s transfer speeds from persistent storage.
+
+While we've optimised the underlying hardware to load models as fast as possible, there are a few things you can do to decrease the time it takes to load your model and, therefore, your coldstart times.
+
+# Using a serialisation and zero-copy initialisation libraries
+
+## Tensorizer (recommended)
+
+[Tensorizer](https://github.com/coreweave/tensorizer) is a library that allows you to load your model from storage into GPU memory in a single step.  
+While initially built to fetch models from S3, it can be used to load models from file as well and so, can be used to load models from Cerebrium's persistent storage which features a near 2GB/s read speed.
+In the case of large models of 20B+ parameters, we've observed a **30-50%** decrease in model loading time which further increases with larger models.  
+For more information on the underlying methods, take a look at their github page [here](https://github.com/coreweave/tensorizer).
+
+In this section below, we'll show you how to use **Tensorizer** to load your model from storage straight into GPU memory in a single step.
+
+### Installation
+
+Add the following to your `requirements.txt` file to install **Tensorizer** in your deployment:
+
+```txt
+tensorizer
+```
+
+### Usage
+
+To use **Tensorizer**, you need to first serialise your model and save it to your persistent-storage.
+
+```python
+from tensorizer import TensorSerializer
+def serialise_model(model, model_path):
+    """Serialise the model and saving the weights to the model_path"""
+    try:
+        serializer = TensorSerializer(model_path)
```

Comment:
Is model path here the HF model path? What if I have my own custom checkpoint file?

Reply:
Save path. Provided your model is loaded, you can tensorize it.
I need to clarify that. Good point.